# Task 9 – Descriptive Statistics Primer


**Dataset:** Titanic Dataset

**Tools:** Python, Pandas, Excel

In [14]:
import pandas as pd

url = "https://raw.githubusercontent.com/datasciencedojo/datasets/master/titanic.csv"

df = pd.read_csv(url)

df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [3]:
df.shape
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [4]:
df.to_csv("Titanic-Dataset.csv", index=False)

In [5]:
df.shape

(891, 12)

In [6]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [8]:
numeric_cols = ["Age", "Fare", "SibSp", "Parch", "Pclass"]

df[numeric_cols].head()

,Age,Fare,SibSp,Parch,Pclass
0,22.0,7.2500,1,0,3
1,38.0,71.2833,1,0,1
2,26.0,7.9250,0,0,3
3,35.0,53.1000,1,0,1
4,35.0,8.0500,0,0,3


In [9]:
df[numeric_cols].describe()

,Age,Fare,SibSp,Parch,Pclass
count,714.000000,891.000000,891.000000,891.000000,891.000000
mean,29.699118,32.204208,0.523008,0.381594,2.308642
std,14.526497,49.693429,1.102743,0.806057,0.836071
min,0.420000,0.000000,0.000000,0.000000,1.000000
25%,20.125000,7.910400,0.000000,0.000000,2.000000
50%,28.000000,14.454200,0.000000,0.000000,3.000000
75%,38.000000,31.000000,1.000000,0.000000,3.000000
max,80.000000,512.329200,8.000000,6.000000,3.000000


In [10]:
df[numeric_cols].mode()

,Age,Fare,SibSp,Parch,Pclass
0,24.0,8.05,0,0,3


In [11]:
summary = pd.DataFrame({
    "Count": df[numeric_cols].count(),
    "Mean": df[numeric_cols].mean(),
    "Median": df[numeric_cols].median(),
    "Mode": df[numeric_cols].mode().iloc[0],
    "Standard Deviation": df[numeric_cols].std(),
    "25th Percentile": df[numeric_cols].quantile(0.25),
    "50th Percentile": df[numeric_cols].quantile(0.50),
    "75th Percentile": df[numeric_cols].quantile(0.75)
})

summary = summary.round(2)

summary

,Count,Mean,Median,Mode,Standard Deviation,25th Percentile,50th Percentile,75th Percentile
Age,714,29.70,28.00,24.00,14.53,20.12,28.00,38.0
Fare,891,32.20,14.45,8.05,49.69,7.91,14.45,31.0
SibSp,891,0.52,0.00,0.00,1.10,0.00,0.00,1.0
Parch,891,0.38,0.00,0.00,0.81,0.00,0.00,0.0
Pclass,891,2.31,3.00,3.00,0.84,2.00,3.00,3.0


## Skewness Analysis

In [12]:
df[numeric_cols].skew()

Age       0.389108
Fare      4.787317
SibSp     3.695352
Parch     2.749117
Pclass   -0.630548
dtype: float64

## Interquartile Range (IQR)

In [13]:
Q1 = df[numeric_cols].quantile(0.25)
Q3 = df[numeric_cols].quantile(0.75)

IQR = Q3 - Q1

IQR.round(2)

Age       17.88
Fare      23.09
SibSp      1.00
Parch      0.00
Pclass     1.00
dtype: float64

## Interpretation

### Age
The mean age is 29.70 years and the median is 28.00 years. The mean is slightly higher than the median, suggesting some positive skewness. The middle 50% of ages range from approximately 20.13 to 38.00 years.

### Fare
The mean fare is 32.20 while the median is 14.45. The higher mean suggests positive/right skewness, with high fare values influencing the average. The standard deviation of 49.69 indicates substantial variability in fares.

### SibSp
The mean is 0.52, while the median and mode are both 0.00. This indicates that having no siblings or spouses aboard was the most common observation.

### Parch
The mean is 0.38, while the median and mode are both 0.00. The 75th percentile is also 0.00, indicating that at least 75% of observations have zero parents or children aboard.

### Pclass
The mean is 2.31, while the median and mode are both 3.00. This indicates that third class is the most frequently occurring passenger class.

## Interview Questions

### 1. When is median a better measure of center than mean?

Median is better when data is skewed or contains extreme values/outliers because it is less affected by unusually high or low observations.

### 2. What does a large standard deviation tell you about a dataset?

A large standard deviation indicates that observations are more widely spread around the mean, showing greater variability in the dataset.